driver-sync/src/main/com/mongodb/client/internal/MongoClientDelegate.java

```diff
@@ -183,7 +185,13 @@ private ClusterDescription getConnectedClusterDescription() {
             WriteBinding binding = getWriteBinding(readConcern, actualClientSession, session == null && actualClientSession != null);
             try {
                 return operation.execute(binding);
-            } finally {
+            } catch (MongoSocketException e) {
+                if (session != null && session.hasActiveTransaction()) {
```

Comment:
This seems really brittle, but I can't think of a better place to add the label.